### PR TITLE
added `--threads-per-core` option (LM SRun)

### PR DESCRIPTION
**Pull request by mtitov**

### CODE REVIEW

File: src/radical/pilot/agent/launch_method/srun.py [hunk 1]
```diff
@@ -142,21 +142,23 @@ def get_launch_cmds(self, task, exec_path):
...
                     + '--ntasks %d '       % n_tasks \
                     + '--cpus-per-task %d' % n_task_threads
 
+        if self._rm_info['threads_per_core'] > 1:
+            mapping += ' --threads-per-core %d' % \
+                       self._rm_info['threads_per_core']
 
         # check that gpus were requested to be allocated
-        if self._rm_info.get('gpus') and n_gpus:
+        if self._rm_info.get('requested_gpus') and n_gpus:
```
Comment (andre-merzky) [inline src/radical/pilot/agent/launch_method/srun.py:161]: nice catch!

**Review by andre-merzky — APPROVED** [review]
LGTM, thanks!

**Comment by mtitov** [comment]
**Update**

While testing these changes on Spock@ORNL, it shows that `--threads-per-core` didn't solve the similar issue there
```
$ cat /gpfs/alpine/scratch/.../task.0000*/task.0000*.err
cpu-bind=MASK - spock25, task  0  0 [20252]: mask 0xff00000000000000ff set
cpu-bind=MASK - spock25, task  1  1 [20253]: mask 0xff00000000000000ff0000 set
srun: Job 470608 step creation temporarily disabled, retrying (Requested nodes are busy)
srun: Step created for job 470608
cpu-bind=MASK - spock25, task  0  0 [20439]: mask 0xff00000000000000ff set
cpu-bind=MASK - spock25, task  1  1 [20440]: mask 0xff00000000000000ff0000 set
```
where `srun` commands are
```
/usr/bin/srun --nodes 1 --ntasks 2 --cpus-per-task 16 --threads-per-core 2 
--gpus-per-task 1 --gpu-bind closest --nodelist=spock25 $RP_TASK_SANDBOX/task.000000.exec.sh
```

These machines use different versions of Slurm
- Spock 	- slurm version: 21.08.7
- Crusher 	- slurm version: 22.05.2

p.s. proposed changes do not create additional issues

**Review by andre-merzky — APPROVED** [review]
LGTM - thanks!

**Pull request merged**